### PR TITLE
Introduce testing for ajax execute and render `@this`

### LINKED ISSUE #20812: Faces 4.0: <f:ajax> execute="@this" and render="@this" does not behave as expected when nested in composite component
User story:
As a user of Open Liberty and the faces-4.0 feature, I want to be able to create asynchronously execute and render nested composite components that work as expected by the specification.  That means that the entire component should be re-rendered, not just the individual form element that was interacted with.  

Known issue: https://issues.apache.org/jira/projects/MYFACES/issues/MYFACES-4505?filter=allopenissues
MyFaces is still executing only individual components within the composite component, due to a known bug.

Spec issue: `https://github.com/jakartaee/faces/issues/1567`

See: https://balusc.omnifaces.org/2021/11/whats-new-in-faces-40.html

### PR DESCRIPTION
**Pull request by KyleAure**
Related to #20812
This introduces tests only, but there is an outstanding myfaces defect that will need to be resolved. 
https://issues.apache.org/jira/projects/MYFACES/issues/MYFACES-4505

### CODE REVIEW

**Comment by KyleAure** [comment]
#build

**Comment by KyleAure** [comment]
#libby check headers
The personal build failed with 3 existing defects.

**Comment by KyleAure** [comment]
#build

**Comment by KyleAure** [comment]
#libby check headers
The personal build failed with one known defect. 
Waiting on review and buildcon to merge

File: dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/AjaxRenderExecuteThisTest.war/resources/resources/components/component.xhtml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <cc:implementation>
+        <div id="#{cc.clientId}">
+            <h:inputText id="input1" value="#{cc.attrs.input1}" />
+            <h:inputText id="input2" value="#{cc.attrs.input2}" />
+            <h:inputText id="input3" value="#{cc.attrs.input3}">
+            	<f:ajax render="@form" />
+            </h:inputText>
+        </div>
+    </cc:implementation>
+</ui:component>
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/AjaxRenderExecuteThisTest.war/resources/resources/components/component.xhtml]: Minor formatting comment, just add a new line here.

File: dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/AjaxRenderExecuteThisTest.war/resources/resources/components/wrapper.xhtml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    xmlns:ui="jakarta.faces.facelets"
+    xmlns:cc="jakarta.faces.composite"
+>
+    <cc:interface />
+    <cc:implementation>
+        <div id="#{cc.clientId}">
+            <cc:insertChildren/>
+        </div>
+    </cc:implementation>
+</ui:component>
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/AjaxRenderExecuteThisTest.war/resources/resources/components/wrapper.xhtml]: Minor formatting comment, just add a new line here.

File: dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/AjaxRenderExecuteThisTest.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    public TestName name = new TestName();
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        WebArchive app = ShrinkHelper.buildDefaultApp(APP_NAME + ".war", "io.openliberty.org.apache.faces40.fat.ajax.beans");
+
+        //Test differences between myfaces and mojarra if needed
+//        app.addAsLibraries(new File("publish/files/mojarra40/").listFiles());
+//        app.addAsDirectories("publish/files/permissions");
```
Comment (pnicolucci) [inline dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/AjaxRenderExecuteThisTest.java:65]: Can you just clean the formatting up on these two lines?

**Review by pnicolucci — COMMENTED** [review]
Overall this looks good, just a couple of minor comments. Thanks!

**Comment by KyleAure** [comment]
#build - rebuild after rebasing
#libby - check headers
@pnicolucci - rereview

**Review by pnicolucci — APPROVED** [review]

**Pull request merged**